#5: Add sensor metadata to messages

glencornell opened the pull request:
This is a rudimentary description of the sensor: the field of view and the orientation (z axis only is compass heading) and the far clipping plane (i.e. max distance to detecting a transmitter).

glencornell commented:
this draft PR does not modify the test files or the documentation to facilitate code review.

christianrowlands commented on src/main/asyncapi/network_survey_messaging.yaml:
```diff
@@ -1665,6 +1825,45 @@ components:
+      type: integer
+      format: float
+      minimum: 0
+      description: The furthest distance in meters that the sensor is capable of detecting a transmitter.
```
Would this field be better indicated by "received sensitivity" in units of dBm? Distance in meters is dependent on a lot of factors so it is hard to specific a value here. However, if we use receiver sensitivity then radio link budget can be used to calculate the distance in meters with the environmental variables. I have a good equation I have used in other code that I can dig up if you are interested.

glencornell replied:
name is changed to receivedSentitivity.  Should the equation/algorithm be embedded in the comments for receivedSensitivity?  If so, let me know and I'll include it.

christianrowlands commented on src/main/asyncapi/network_survey_messaging.yaml:
```diff
@@ -1665,6 +1825,45 @@ components:
+      format: float
+      minimum: -180.0
+      maximum: 180.0
+      description: The rotation reading contains 3 angles, measured in degrees that define the orientation of the device in three-dimensional space. These angles are similar to yaw, pitch and roll but are defined using only right hand rotation with axes as defined by the right hand cartesian coordinate system. The three angles are applied to the device in the following order. Right-handed rotatio ...
```
For all of these rotation fields, it might be best to include that this is for the antenna. The description is great and is very detailed on the specifics of this value, but it would be helpful to include in the first sentence or two a high level view of "Why would I want to use this value".

glencornell replied:
I changed this to sensor/antenna and added the "so what".

christianrowlands commented on src/main/proto/com/craxiom/messaging/cdma_record.proto:
```diff
@@ -31,6 +31,15 @@ message CdmaRecordData {
+    float rotationX = 50;
+    float rotationY = 51;
+    float rotationZ = 52;
+    bool directionalSensor = 53;
```
I don't see the `directionalSensor` field in the AsyncAPI yaml spec. Is this needed since we have the field of view?

glencornell replied:
directionalSensor was a vestige of an older iteration.  I have eliminated all references to it.

christianrowlands reviewed (commented):
Sorry for taking WAY to long on this, but I think I see where you are coming from now. The only other thing that would be good is if you wrote some unit tests for the json conversions.

glencornell commented:
FYI - Some additional pork in this pull request:
1. I regenerated the html documents.
2. I added a python generator for the protobuf code (the build.gradle needed updating to support this).
3. I updated the unit tests and ran `./gradlew clean build test`

christianrowlands commented on docs/CNAME:
```diff
@@ -1 +0,0 @@
-messaging.networksurvey.app
```
This file is needed for the GitHub Pages to allow for a CNAME.

christianrowlands commented on build.gradle:
```diff
@@ -31,8 +33,8 @@ repositories {
 }
 
 ext {
-    grpcVersion = '1.46.0'
```
Did you have to revert these version numbers for the python lib generation to work?

christianrowlands reviewed (approved):
I will add the CNAME file back in after it is merged. I am guessing there is something wrong with the HTML generation code.
